prusti-tests/tests/verify_overflow/pass/competitve_programming/dp/TheJackpot_10684.rs

```diff
@@ -0,0 +1,120 @@
+//https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&category=24&page=show_problem&problem=1625
+use prusti_contracts::*;
+
+pub struct VecWrapperI32 {
+    v: Vec<i32>,
+}
+
+impl VecWrapperI32 {
+    #[trusted]
+    #[pure]
+    #[ensures (0 <= result)]
+    pub fn len(&self) -> usize {
+        self.v.len()
+    }
+
+    #[trusted]
+    #[ensures (result.len() == 0)]
+    pub fn new() -> Self {
+        VecWrapperI32 { v: Vec::new() }
+    }
+
+    #[trusted]
+    #[pure]
+    #[requires (0 <= index && index < self.len())]
+    pub fn lookup(&self, index: usize) -> i32 {
+        self.v[index]
+    }
+
+    #[trusted]
+    #[ensures (self.len() == old(self.len()) + 1)]
+    #[ensures (self.lookup(old(self.len())) == value)]
+    #[ensures (forall(|i: usize| (0 <= i && i < old(self.len())) ==>
+                    self.lookup(i) == old(self.lookup(i))))]
+    pub fn push(&mut self, value: i32) {
+        self.v.push(value);
+    }
+}
+
+#[pure]
+#[ensures (result >= a && result >= b)]
+#[ensures (result == a || result == b)]
+fn max(a: i32, b: i32) -> i32 {
+    if a < b {
+        b
+    } else {
+        a
+    }
+}
+
+#[pure]
+#[requires (i >= 0)]
+#[requires (i < seq.len())]
+#[requires (seq.len() > 0)]
+fn solve_rec(seq: &VecWrapperI32, i: usize) -> i32 {
```

Comment:
I would say that it is not obvious that this function is computing the result we want it to compute. Therefore, we need to prove that this is indeed the case. One idea would be to do this as follows:

1. Write a function `solve_naive(seq: &VecWrapperI32) -> i32` that solves the problem by using two nested loops (one loop checks all possible start positions of the subsequence, and the second one computes all possible sums that start at that position).
2. Prove that `solve_rec(seq, seq.len()) >= solve_naive(seq)` (it should be much easier to prove `>=` than `==`).

Is it clear how to do that?

Reply:
I will try to do it and see if I manage.

Reply:
If you get any questions, please let me know.